tools/deviceGen/output/traits_defaults.hlsl

```diff
@@ -0,0 +1,382 @@
+NBL_GENERATE_GET_OR_DEFAULT(shaderDenormFlushToZeroFloat32, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderDenormFlushToZeroFloat64, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderRoundingModeRTEFloat16, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderRoundingModeRTEFloat32, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderRoundingModeRTEFloat64, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderRoundingModeRTZFloat16, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderRoundingModeRTZFloat32, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderRoundingModeRTZFloat64, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(maxUpdateAfterBindDescriptorsInAllPools, uint32_t, 1048576);
+NBL_GENERATE_GET_OR_DEFAULT(shaderUniformBufferArrayNonUniformIndexingNative, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderSampledImageArrayNonUniformIndexingNative, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderStorageBufferArrayNonUniformIndexingNative, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderStorageImageArrayNonUniformIndexingNative, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(shaderInputAttachmentArrayNonUniformIndexingNative, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(robustBufferAccessUpdateAfterBind, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(quadDivergentImplicitLod, bool, false);
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageDescriptorUpdateAfterBindSamplers, uint32_t, 500000);
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageDescriptorUpdateAfterBindUBOs, uint32_t, 15);
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageDescriptorUpdateAfterBindSSBOs, uint32_t, 500000);
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageDescriptorUpdateAfterBindImages, uint32_t, 500000);
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageDescriptorUpdateAfterBindStorageImages, uint32_t, 500000);
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageDescriptorUpdateAfterBindInputAttachments, uint32_t, MinMaxColorAttachments);
+NBL_GENERATE_GET_OR_DEFAULT(maxPerStageUpdateAfterBindResources, uint32_t, 500000);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetUpdateAfterBindSamplers, uint32_t, 500000);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetUpdateAfterBindUBOs, uint32_t, 72);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetUpdateAfterBindDynamicOffsetUBOs, uint32_t, 8);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetUpdateAfterBindSSBOs, uint32_t, 500000);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetUpdateAfterBindDynamicOffsetSSBOs, uint32_t, 4);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetUpdateAfterBindImages, uint32_t, 500000);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetUpdateAfterBindStorageImages, uint32_t, 500000);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetUpdateAfterBindInputAttachments, uint32_t, MinMaxColorAttachments);
+NBL_GENERATE_GET_OR_DEFAULT(supportedDepthResolveModes, core::bitflag<RESOLVE_MODE_FLAGS>, RESOLVE_MODE_FLAGS::SAMPLE_ZERO_BIT);
+NBL_GENERATE_GET_OR_DEFAULT(supportedStencilResolveModes, core::bitflag<RESOLVE_MODE_FLAGS>, RESOLVE_MODE_FLAGS::SAMPLE_ZERO_BIT);
```

Comment:
remember about using the enums directly, migrating them to HLSL header and having them unscoped